v2/tools/generator/internal/astmodel/interface_type.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"go/token"
+	"sort"
+	"strings"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astbuilder"
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/readonly"
+)
+
+var EmptyInterfaceType = NewInterfaceType()
+
+// InterfaceType represents an interface
+type InterfaceType struct {
+	functions readonly.Map[string, Function]
+}
+
+var _ Type = &InterfaceType{}
+
+// NewInterfaceType creates a new InterfaceType
+func NewInterfaceType() *InterfaceType {
```

Comment:
Suggest having a varidic parameter allowing functions to be added straight away:

```suggestion
func NewInterfaceType(functions ...Function) *InterfaceType {
```